react/finger-exercises/redux/02 - react-redux/src/redux/book/reducer.js

```diff
@@ -28,7 +28,15 @@ function reducer(state = initialState, action) {
     case actions.REMOVE_ITEM:
       return {
         ...state,
-        bookSelected: state.bookSelected.filter(book => book.id !== action.payload)
+        bookSelected: state.bookSelected.reduce((accumulator, currentValue) => {
+          if (currentValue.quantity > 1 && currentValue.id === action.payload) {
+            return [...accumulator, { ...currentValue, quantity: currentValue.quantity - 1 }];
```

Comment:
i'd do `accumulator.push()`, because the idea of the reduce is to add things to the accumulator, and here you are creating a new array every iteration,
 also you could do
```
if(current.id === action.payload) {
  const newQuantity = current.quantity - 1;
  if(newQuantity > 0) accum.push({...currentValue, quantity: newQuantity})
} else {
  accum.push(currentValue);
}
 return accum;
```

what do you think?